code/python/segment_chlamy.v.2.1.py

```diff
@@ -0,0 +1,168 @@
+    # Find the starting and ending positions for the substring we want to extract
+    start_index = filename.find('pools_') + len('pools_')
+    end_index = filename.rfind('_seq')
+    # If the start and end positions are valid, extract the substring; otherwise, set to None
+    metadata_pool_id = filename[start_index:end_index] if start_index != -1 and end_index != -1 else None
+
+    # Extract the 'frames' from the filename in a similar manner as above
+    start_index = filename.find("_f") + 2
+    end_index = filename.find("_Prob")
+    metadata_frames = filename[start_index:end_index] if start_index != -1 and end_index != -1 else None
+
+    # Extract the 'sequence' metadata
+    start_index = filename.find("seq") + 3
+    end_index = filename.find("_f")
+    metadata_seq = filename[start_index:end_index] if start_index != -1 and end_index != -1 else None
+
+    # Return a dictionary containing all extracted metadata
+    return {
+        'metadata_experiment': metadata_experiment,
+        'metadata_species': metadata_species,
+        'metadata_pool_id': metadata_pool_id,
+        'metadata_frames': metadata_frames,
+        'metadata_sequence': metadata_seq
+    }
+
+# Function to process a directory containing image files
+def process_directory(root_directory):
+    # Loop through each sub-directory and file in the root directory
+    for root, dirs, files in os.walk(root_directory):
+        # Only process directories that contain 'prob_maps_org' in their name
+        if 'prob_maps_org' in root:
+            # Loop through each file in the directory
+            for filename in files:
+                # Only process files with the '.tif' extension
+                if filename.endswith('.tif'):
+                    # Create the full path to the input file
+                    input_path = os.path.join(root, filename)
+                    # Create the directory path where the output will be saved, replacing 'prob_maps_organized' with 'objects'
+                    output_dir = root.replace('prob_maps_organized', 'objects')
+                    # If the output directory doesn't exist, create it
+                    if not os.path.exists(output_dir):
+                        os.makedirs(output_dir)
+                    # Create the full path to the output file
+                    output_path = os.path.join(output_dir, filename)
+
+                    # Run the cell segmentation function on the input file
+                    binary_map, properties = segment_cells(input_path)
+                    # Save the segmented image
+                    imsave(output_path, binary_map)
+                    # Save the properties of the segmented cells to a CSV file
+                    save_measurements_to_csv(input_path, properties)
+
+# Main function, entry point of the script
+if __name__ == "__main__":
+    # Starting directory (change this to your specific directory if needed)
+    base_directory = "."
```

Comment:
This doesn't have to be experiments?

Reply:
This is correct. It doesn't have to be experiments